components/camel-fhir/src/main/java/org/apache/camel/component/fhir/FhirProducer.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.fhir;
+
+import java.util.List;
+import ca.uhn.fhir.model.base.resource.BaseOperationOutcome;
+import ca.uhn.fhir.rest.gclient.IClientExecutable;
+import org.apache.camel.Exchange;
+import org.apache.camel.impl.DefaultProducer;
+
+/**
+ * The fhir producer. It simply retrieves a {@link IClientExecutable}, executes it and sets the result in the Camel exchange body.
+ */
+public class FhirProducer extends DefaultProducer {
+    
+    public FhirProducer(FhirEndpoint endpoint) {
+        super(endpoint);
+    }
+
+    public void process(Exchange exchange) throws Exception {
+        IClientExecutable clientExecutable = (IClientExecutable) exchange.getIn().getHeaders().get(FhirConstants.FHIR_REQUEST);
```

Comment:
This seems a bit odd to use a header for such a thing. How will this header to be configured? Headers should be part of the message contract, and not "misused" for client stuff

Reply:
I'll move this in the message body